src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -80,8 +80,18 @@ bool Camera::on_render_begin(
     IAbortSwitch*           abort_switch)
 {
     m_shutter_open_time = m_params.get_optional<float>("shutter_open_time", 0.0f);
+    m_shutter_open_end_time = m_params.get_optional<float>("shutter_open_end_time", 0.0f);
+    m_shutter_close_start_time = m_params.get_optional<float>("shutter_close_start_time", 1.0f);
     m_shutter_close_time = m_params.get_optional<float>("shutter_close_time", 1.0f);
     m_shutter_open_time_interval = m_shutter_close_time - m_shutter_open_time;
+    m_normalized_open_end_time = lerp2(m_shutter_open_time, m_shutter_close_time, m_shutter_open_end_time);
+    m_normalized_open_end_time_half = m_normalized_open_end_time / 2;
+    m_normalized_close_start_time = lerp2(m_shutter_open_time, m_shutter_close_time, m_shutter_close_start_time);
+    m_height = 2 / (1 + m_normalized_close_start_time - m_normalized_open_end_time);
```

Comment:
We need to find a better name than `m_height`...

Reply:
How about... `shutter_pdf_max_height`?

Reply:
Sounds good!